### PR TITLE
Docs

### PR DESCRIPTION
**Pull request by skryzhny**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,81 @@
 # git-keeper
+
+## Purpose
+This utility is ment to backup git repositories.
+Repositories are clonned in `--mirror` mode, `tar`-ed and `gpg` encrypted, then uploaded to S3 bucket
```
Comment (jmelis) [inline README.md]: typo `cloned`

Comment (jmelis) [reply README.md]: typo `to an S3 bucket.`

File: README.md [hunk 2]
```diff
@@ -1 +1,81 @@
...
+
+## Simple example of using docker image
+```
+cat repolist | docker run --rm -i -v /home/user/github/git-keeper/config:/config:z \
```
Comment (jmelis) [inline README.md]: for clarity perhaps `-v $PWD/config:/config:z`

File: README.md [hunk 3]
```diff
@@ -1 +1,81 @@
...
+Version: GnuPG v2.0.22 (GNU/Linux)
+
+mQ.....
+....
+
+=qwert
+
+-----END PGP PUBLIC KEY BLOCK-----
+```
+Optional `--subfolders` parameter can take several pathes.
```
Comment (jmelis) [inline README.md]: typo `paths`

File: README.md [hunk 4]
```diff
@@ -1 +1,81 @@
...
+This is useful if you want put same backups in different folders with different retention policies.
+
+### config/.netrc file
+There is some hack for backing up private repos. You need to put credentials for private repos to it like:
+```
+machine gitlab.your.company.com
+login automated-backups
+password somesecretword
+```
+image contain symlink to /config.netrc file so content will be used by console `git` command when clonning private repos.
```
Comment (jmelis) [inline README.md]: typo `clonning`

Comment (jmelis) [reply README.md]: typo: `/config/.netrc`

Comment (jmelis) [reply README.md]: typo `Image contains a symlink ...`

File: README.md [hunk 5]
```diff
@@ -1 +1,81 @@
 # git-keeper
+
+## Purpose
+This utility is ment to backup git repositories.
```
Comment (jmelis) [inline README.md]: typo `ment`

File: README.md [hunk 6]
```diff
@@ -1 +1,81 @@
...
+## Simple example of using docker image
+```
+cat repolist | docker run --rm -i -v /home/user/github/git-keeper/config:/config:z \
+    -e GIT_SSL_NO_VERIFY=true \
+    quay.io/app-sre/git-keeper:latest \
+    --config /config/gk_tomls \
+    --gpgs /config/gpg \
+    --subfolders backups/weeklytest1,backups/weeklytest2,backups/daily
+```
+
```
Comment (jmelis) [inline README.md]: perhaps we could talk a bit about the structure of `repolist` file (just one repo per line)

File: README.md [hunk 7]
```diff
@@ -1 +1,81 @@
...
+There is some hack for backing up private repos. You need to put credentials for private repos to it like:
+```
+machine gitlab.your.company.com
+login automated-backups
+password somesecretword
+```
+image contain symlink to /config.netrc file so content will be used by console `git` command when clonning private repos.
+
+## Usage
+Utility is supposed to run inside docker container and accept list of repositories to backup via pipe
```
Comment (jmelis) [inline README.md]: I'd move the simple example here

File: README.md [hunk 8]
```diff
@@ -1 +1,81 @@
...
+
+## building/testing
+For testing python parts with tox just `make test`
+
+For building localy `make build`
+
+`make push` will push to your image registry
```
Comment (jmelis) [inline README.md]: I'd remove this as we have hardcoded the path: https://github.com/app-sre/git-keeper/blob/master/Makefile#L2

File: README.md [hunk 9]
```diff
@@ -1 +1,81 @@
...
+```
+image contain symlink to /config.netrc file so content will be used by console `git` command when clonning private repos.
+
+## Usage
+Utility is supposed to run inside docker container and accept list of repositories to backup via pipe
+
+## building/testing
+For testing python parts with tox just `make test`
+
+For building localy `make build`
```
Comment (jmelis) [inline README.md]: we should specify what container this outputs (even if it can be inferred from the simple example)

File: README.md [hunk 10]
```diff
@@ -1 +1,81 @@
...
+
+`make push` will push to your image registry
+
+## Future enchantments
```
Comment (jmelis) [inline README.md]: typo `enhancements`

File: README.md [hunk 11]
```diff
@@ -1 +1,81 @@
...
+- Using host PKI folder so no need to set `-e GIT_SSL_NO_VERIFY=true` for docker run command
+- Find a way to restore pull-request to github/gitlab
+- Backup issues, wiki pages
+- More error handling/reporting, tests
+- ?
```
Comment (jmelis) [inline README.md]: remove this line

File: README.md [hunk 12]
```diff
@@ -1 +1,81 @@
...
+
+For building localy `make build`
+
+`make push` will push to your image registry
+
+## Future enchantments
+- Using host PKI folder so no need to set `-e GIT_SSL_NO_VERIFY=true` for docker run command
+- Find a way to restore pull-request to github/gitlab
+- Backup issues, wiki pages
+- More error handling/reporting, tests
```
Comment (jmelis) [inline README.md]: what in particular? otherwise remove

File: README.md [hunk 13]
```diff
@@ -1 +1,81 @@
...
+For testing python parts with tox just `make test`
+
+For building localy `make build`
+
+`make push` will push to your image registry
+
+## Future enchantments
+- Using host PKI folder so no need to set `-e GIT_SSL_NO_VERIFY=true` for docker run command
+- Find a way to restore pull-request to github/gitlab
+- Backup issues, wiki pages
```
Comment (jmelis) [inline README.md]: not sure this is in the scope of this project. wdyt @maorfr ?

File: README.md (hunk 12 above)
Comment (skryzhny) [reply README.md]: it's more about better logging

**Pull request merged**